src/client/mod.rs

```diff
@@ -536,3 +561,129 @@ impl WeakClient {
+                .insert(provider, tls_options);
+        }
+        EncryptedClientBuilder {
+            options: self.options,
+            enc_options: self.enc_options,
+        }
+    }
+
+    /// Add configuration for multiple KMS providers.  Already-added configuration for other
+    /// providers will be preserved.
+    ///
+    /// ```no_run
+    /// # use bson::doc;
+    /// # use mongocrypt::ctx::KmsProvider;
+    /// # use mongodb::Client;
+    /// # let client_options = todo!();
+    /// # let kv_namespace = todo!();
+    /// # let local_key = doc! { };
+    /// let client = Client::encrypted_builder(client_options, kv_namespace)
+    ///     .extend_kms_providers([
+    ///         (KmsProvider::Local, doc! { "key": local_key }),
+    ///         (KmsProvider::Kmip, doc! { "endpoint": "localhost:5698" }),
+    ///      ])
+    ///     .build();
+    /// ```
+    pub fn extend_kms_providers(
+        mut self,
+        providers: impl IntoIterator<Item = (KmsProvider, bson::Document)>,
+    ) -> EncryptedClientBuilder<{ client_encryption::BUILDER_READY }> {
+        self.enc_options.kms_providers.extend(providers);
+        EncryptedClientBuilder {
+            options: self.options,
+            enc_options: self.enc_options,
+        }
+    }
+
+    pub fn key_vault_client(mut self, client: impl Into<Option<Client>>) -> Self {
+        self.enc_options.key_vault_client = client.into();
+        self
+    }
+
+    pub fn schema_map(mut self, map: impl IntoIterator<Item = (String, Document)>) -> Self {
+        self.enc_options.schema_map = Some(map.into_iter().collect());
+        self
+    }
+
+    pub fn bypass_auto_encryption(mut self, bypass: impl Into<Option<bool>>) -> Self {
+        self.enc_options.bypass_auto_encryption = bypass.into();
+        self
+    }
+
+    pub fn extra_options(mut self, extra_opts: impl Into<Option<Document>>) -> Self {
+        self.enc_options.extra_options = extra_opts.into();
+        self
+    }
+
+    /// Add TLS options for multiple KMS providers.  Already-added configuration for other providers
+    /// will be preserved.
+    pub fn extend_tls_options(
+        mut self,
+        options: impl IntoIterator<Item = (KmsProvider, TlsOptions)>,
+    ) -> Self {
+        self.enc_options
+            .tls_options
+            .get_or_insert_with(KmsProvidersTlsOptions::new)
+            .extend(options);
+        self
+    }
+
+    pub fn encrypted_fields_map(
+        mut self,
+        map: impl IntoIterator<Item = (String, Document)>,
+    ) -> Self {
+        self.enc_options.encrypted_fields_map = Some(map.into_iter().collect());
+        self
+    }
+
+    pub fn bypass_query_analysis(mut self, bypass: impl Into<Option<bool>>) -> Self {
+        self.enc_options.bypass_query_analysis = bypass.into();
+        self
+    }
+}
+
+#[cfg(feature = "csfle")]
+impl EncryptedClientBuilder<{ client_encryption::BUILDER_READY }> {
+    pub async fn build(self) -> Result<Client> {
```

Comment:
Think it would be good to note in this docstring what kind of work this method does, since I think it is a little bit uncommon (but not an issue) for builder's `build` methods to be async.